File: leetcode/字符串/javascript/addbinary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function (a, b) {
    //return (Number.parseInt(a, 2) + Number.parseInt(b, 2)).toString(2);
    var minLen = Math.min(a.length, b.length) - 1;
    var maxLen = Math.max(a.length, b.length) - 1;
    var strS = a.length >= b.length ? b : a;
    var strB = a.length >= b.length ? a : b;
    var c = [];
    var carry = 0;
    while (minLen > -1) {
        var temp = parseInt(strB[maxLen--]) + parseInt(strS[minLen--]) + carry;
        switch (temp) {
            case 0:
                c.push('0');
                break;
            case 1:
                c.push('1');
                carry = 0;
                break;
            case 2:
                c.push('0');
                carry = 1;
                break;
            case 3:
                c.push('1');
                carry = 1;
                break;
        }
    }
    while (maxLen > -1 && carry){
        var temp = parseInt(strB[maxLen--]) + carry;
        if (temp === 1){
            c.push('1');
            carry = 0;
        } else {
            c.push('0');
            carry = 1;
        }
    }
    if (carry)
        c.push('1');
    c.reverse();
    return (maxLen > -1 ? strB.slice(0, maxLen + 1) : '') + c.join('');
};

console.log(addBinary('0', '0'));

File: leetcode/字符串/python/addbinary.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        minLen = min(len(a), len(b)) - 1
        maxLen = max(len(a), len(b)) - 1
        strS = b if len(a) >= len(b) else a
        strB = a if len(a) >= len(b) else b
        c, carry = [], 0
        while minLen > -1:
            temp = int(strB[maxLen]) + int(strS[minLen]) + carry
            if temp == 0:
                c.append('0')
            elif temp == 1:
                c.append('1')
                carry = 0
            elif temp == 2:
                c.append('0')
                carry = 1
            else:
                c.append('1')
                carry = 1
            maxLen -= 1
            minLen -= 1
            
        while (maxLen > -1 and carry):
            temp = int(strB[maxLen]) + carry
            if temp == 1:
                c.append('1')
                carry = 0
            else:
                c.append('0')
                carry = 1
